static/app/components/assistant/guideAnchor.tsx

```diff
@@ -96,6 +96,7 @@ class BaseGuideAnchor extends Component<Props, State> {
   }
 
   // TODO(TS): Reflux returns "Function" instead of () => void
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
   unsubscribe: Function | undefined;
```

Comment:
@TkDodo wdyt about augmenting reflux to change the type?

Reply:
I tried that but it doesn’t really work. If we use module augmentation and overwrite the store in `reflux.d.ts` to:

```
interface Store {
  listen(callback: () => void, bindContext: any): () => void;
}
```

`Store.listen` will basically get an additional overload, and `Function | () => void` will just be collapsed again to `Function`

I think our only chance would be `patch-package`, but I don’t think it’s worth it.

Reply:
ooh but we can just do this:

```
unsubscribe: ReturnType<typeof GuideStore.listen> | undefined;
```

not sure why I didn’t think of this before 😅 

db550d74

Reply:
Agh, that overload doesnt make sense then yes. ReturnType is 👌🏼